Description: Fixing more IDO schema problems
Bug: https://github.com/Icinga/icinga2/issues/4950
Author: Markus Frosch <lazyfrosch@debian.org>
Last-Update: 2017-01-24
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/lib/db_ido_mysql/schema/mysql.sql
+++ b/lib/db_ido_mysql/schema/mysql.sql
@@ -24,7 +24,7 @@
 CREATE TABLE IF NOT EXISTS icinga_acknowledgements (
   acknowledgement_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   entry_time_usec  int default 0,
   acknowledgement_type smallint default 0,
   object_id bigint unsigned default 0,
@@ -34,7 +34,7 @@
   is_sticky smallint default 0,
   persistent_comment smallint default 0,
   notify_contacts smallint default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   PRIMARY KEY  (acknowledgement_id)
 ) ENGINE=InnoDB COMMENT='Current and historical host and service acknowledgements';
 
@@ -64,20 +64,20 @@
 CREATE TABLE IF NOT EXISTS icinga_commenthistory (
   commenthistory_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   entry_time_usec  int default 0,
   comment_type smallint default 0,
   entry_type smallint default 0,
   object_id bigint unsigned default 0,
-  comment_time timestamp NULL DEFAULT NULL,
+  comment_time timestamp NULL,
   internal_comment_id bigint unsigned default 0,
   author_name varchar(64) character set latin1  default '',
   comment_data TEXT character set latin1,
   is_persistent smallint default 0,
   comment_source smallint default 0,
   expires smallint default 0,
-  expiration_time timestamp NULL DEFAULT NULL,
-  deletion_time timestamp NULL DEFAULT NULL,
+  expiration_time timestamp NULL,
+  deletion_time timestamp NULL,
   deletion_time_usec  int default 0,
   name TEXT character set latin1 default NULL,
   PRIMARY KEY  (commenthistory_id),
@@ -93,19 +93,19 @@
 CREATE TABLE IF NOT EXISTS icinga_comments (
   comment_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   entry_time_usec  int default 0,
   comment_type smallint default 0,
   entry_type smallint default 0,
   object_id bigint unsigned default 0,
-  comment_time timestamp NULL DEFAULT NULL,
+  comment_time timestamp NULL,
   internal_comment_id bigint unsigned default 0,
   author_name varchar(64) character set latin1  default '',
   comment_data TEXT character set latin1,
   is_persistent smallint default 0,
   comment_source smallint default 0,
   expires smallint default 0,
-  expiration_time timestamp NULL DEFAULT NULL,
+  expiration_time timestamp NULL,
   name TEXT character set latin1 default NULL,
   session_token int default NULL,
   PRIMARY KEY  (comment_id),
@@ -156,11 +156,11 @@
   disposition varchar(32) character set latin1  default '',
   connect_source varchar(32) character set latin1  default '',
   connect_type varchar(32) character set latin1  default '',
-  connect_time timestamp NULL DEFAULT NULL,
-  disconnect_time timestamp NULL DEFAULT NULL,
-  last_checkin_time timestamp NULL DEFAULT NULL,
-  data_start_time timestamp NULL DEFAULT NULL,
-  data_end_time timestamp NULL DEFAULT NULL,
+  connect_time timestamp NULL,
+  disconnect_time timestamp NULL,
+  last_checkin_time timestamp NULL,
+  data_start_time timestamp NULL,
+  data_end_time timestamp NULL,
   bytes_processed bigint unsigned  default '0',
   lines_processed bigint unsigned  default '0',
   entries_processed bigint unsigned  default '0',
@@ -208,9 +208,9 @@
   contactnotificationmethod_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   contactnotification_id bigint unsigned default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   command_object_id bigint unsigned default 0,
   command_args TEXT character set latin1,
@@ -229,9 +229,9 @@
   instance_id bigint unsigned default 0,
   notification_id bigint unsigned default 0,
   contact_object_id bigint unsigned default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   PRIMARY KEY  (contactnotification_id),
   UNIQUE KEY instance_id (instance_id,contact_object_id,start_time,start_time_usec)
@@ -282,11 +282,11 @@
   contactstatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   contact_object_id bigint unsigned default 0,
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   host_notifications_enabled smallint default 0,
   service_notifications_enabled smallint default 0,
-  last_host_notification timestamp NULL DEFAULT NULL,
-  last_service_notification timestamp NULL DEFAULT NULL,
+  last_host_notification timestamp NULL,
+  last_service_notification timestamp NULL,
   modified_attributes  int default 0,
   modified_host_attributes  int default 0,
   modified_service_attributes  int default 0,
@@ -357,7 +357,7 @@
   customvariablestatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   object_id bigint unsigned default 0,
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   has_been_modified smallint default 0,
   varname varchar(255) character set latin1 collate latin1_general_cs default NULL,
   varvalue TEXT character set latin1,
@@ -377,8 +377,8 @@
   dbversion_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   name varchar(10) character set latin1  default '',
   version varchar(10) character set latin1  default '',
-  create_time timestamp NULL DEFAULT NULL,
-  modify_time timestamp NULL DEFAULT NULL,
+  create_time timestamp NULL,
+  modify_time timestamp NULL,
   PRIMARY KEY (dbversion_id),
   UNIQUE KEY dbversion (name)
 ) ENGINE=InnoDB;
@@ -394,23 +394,23 @@
   instance_id bigint unsigned default 0,
   downtime_type smallint default 0,
   object_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   author_name varchar(64) character set latin1  default '',
   comment_data TEXT character set latin1,
   internal_downtime_id bigint unsigned default 0,
   triggered_by_id bigint unsigned default 0,
   is_fixed smallint default 0,
   duration bigint(20) default 0,
-  scheduled_start_time timestamp NULL DEFAULT NULL,
-  scheduled_end_time timestamp NULL DEFAULT NULL,
+  scheduled_start_time timestamp NULL,
+  scheduled_end_time timestamp NULL,
   was_started smallint default 0,
-  actual_start_time timestamp NULL DEFAULT NULL,
+  actual_start_time timestamp NULL,
   actual_start_time_usec  int default 0,
-  actual_end_time timestamp NULL DEFAULT NULL,
+  actual_end_time timestamp NULL,
   actual_end_time_usec  int default 0,
   was_cancelled smallint default 0,
   is_in_effect smallint default 0,
-  trigger_time timestamp NULL DEFAULT NULL,
+  trigger_time timestamp NULL,
   name TEXT character set latin1 default NULL,
   PRIMARY KEY  (downtimehistory_id),
   UNIQUE KEY instance_id (instance_id,object_id,entry_time,internal_downtime_id)
@@ -429,9 +429,9 @@
   object_id bigint unsigned default 0,
   state smallint default 0,
   state_type smallint default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   command_object_id bigint unsigned default 0,
   command_args TEXT character set latin1,
@@ -455,7 +455,7 @@
 CREATE TABLE IF NOT EXISTS icinga_externalcommands (
   externalcommand_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   command_type smallint default 0,
   command_name varchar(128) character set latin1  default '',
   command_args TEXT character set latin1,
@@ -471,7 +471,7 @@
 CREATE TABLE IF NOT EXISTS icinga_flappinghistory (
   flappinghistory_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  event_time timestamp NULL DEFAULT NULL,
+  event_time timestamp NULL,
   event_time_usec  int default 0,
   event_type smallint default 0,
   reason_type smallint default 0,
@@ -480,7 +480,7 @@
   percent_state_change double  default '0',
   low_threshold double  default '0',
   high_threshold double  default '0',
-  comment_time timestamp NULL DEFAULT NULL,
+  comment_time timestamp NULL,
   internal_comment_id bigint unsigned default 0,
   PRIMARY KEY  (flappinghistory_id)
 ) ENGINE=InnoDB  COMMENT='Current and historical record of host and service flapping';
@@ -501,9 +501,9 @@
   max_check_attempts smallint default 0,
   state smallint default 0,
   state_type smallint default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   command_object_id bigint unsigned default 0,
   command_args TEXT character set latin1,
@@ -709,7 +709,7 @@
   hoststatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   host_object_id bigint unsigned default 0,
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   output TEXT character set latin1,
   long_output TEXT,
   perfdata TEXT character set latin1,
@@ -719,18 +719,18 @@
   should_be_scheduled smallint default 0,
   current_check_attempt smallint default 0,
   max_check_attempts smallint default 0,
-  last_check timestamp NULL DEFAULT NULL,
-  next_check timestamp NULL DEFAULT NULL,
+  last_check timestamp NULL,
+  next_check timestamp NULL,
   check_type smallint default 0,
-  last_state_change timestamp NULL DEFAULT NULL,
-  last_hard_state_change timestamp NULL DEFAULT NULL,
+  last_state_change timestamp NULL,
+  last_hard_state_change timestamp NULL,
   last_hard_state smallint default 0,
-  last_time_up timestamp NULL DEFAULT NULL,
-  last_time_down timestamp NULL DEFAULT NULL,
-  last_time_unreachable timestamp NULL DEFAULT NULL,
+  last_time_up timestamp NULL,
+  last_time_down timestamp NULL,
+  last_time_unreachable timestamp NULL,
   state_type smallint default 0,
-  last_notification timestamp NULL DEFAULT NULL,
-  next_notification timestamp NULL DEFAULT NULL,
+  last_notification timestamp NULL,
+  next_notification timestamp NULL,
   no_more_notifications smallint default 0,
   notifications_enabled smallint default 0,
   problem_has_been_acknowledged smallint default 0,
@@ -824,8 +824,8 @@
 CREATE TABLE IF NOT EXISTS icinga_logentries (
   logentry_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  logentry_time timestamp NULL DEFAULT NULL,
-  entry_time timestamp NULL DEFAULT NULL,
+  logentry_time timestamp NULL,
+  entry_time timestamp NULL,
   entry_time_usec  int default 0,
   logentry_type  int default 0,
   logentry_data TEXT character set latin1,
@@ -847,9 +847,9 @@
   notification_type smallint default 0,
   notification_reason smallint default 0,
   object_id bigint unsigned default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   state smallint default 0,
   output TEXT character set latin1,
@@ -887,7 +887,7 @@
   processevent_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   event_type smallint default 0,
-  event_time timestamp NULL DEFAULT NULL,
+  event_time timestamp NULL,
   event_time_usec  int default 0,
   process_id bigint unsigned default 0,
   program_name varchar(16) character set latin1  default '',
@@ -906,17 +906,17 @@
   programstatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   program_version varchar(64) character set latin1 collate latin1_general_cs default NULL,
-  status_update_time timestamp NULL DEFAULT NULL,
-  program_start_time timestamp NULL DEFAULT NULL,
-  program_end_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
+  program_start_time timestamp NULL,
+  program_end_time timestamp NULL,
   endpoint_name varchar(255) character set latin1 collate latin1_general_cs default NULL,
   is_currently_running smallint default 0,
   process_id bigint unsigned default 0,
   daemon_mode smallint default 0,
-  last_command_check timestamp NULL DEFAULT NULL,
-  last_log_rotation timestamp NULL DEFAULT NULL,
+  last_command_check timestamp NULL,
+  last_log_rotation timestamp NULL,
   notifications_enabled smallint default 0,
-  disable_notif_expire_time timestamp NULL DEFAULT NULL,
+  disable_notif_expire_time timestamp NULL,
   active_service_checks_enabled smallint default 0,
   passive_service_checks_enabled smallint default 0,
   active_host_checks_enabled smallint default 0,
@@ -961,20 +961,20 @@
   instance_id bigint unsigned default 0,
   downtime_type smallint default 0,
   object_id bigint unsigned default 0,
-  entry_time timestamp NULL DEFAULT NULL,
+  entry_time timestamp NULL,
   author_name varchar(64) character set latin1  default '',
   comment_data TEXT character set latin1,
   internal_downtime_id bigint unsigned default 0,
   triggered_by_id bigint unsigned default 0,
   is_fixed smallint default 0,
   duration bigint(20) default 0,
-  scheduled_start_time timestamp NULL DEFAULT NULL,
-  scheduled_end_time timestamp NULL DEFAULT NULL,
+  scheduled_start_time timestamp NULL,
+  scheduled_end_time timestamp NULL,
   was_started smallint default 0,
-  actual_start_time timestamp NULL DEFAULT NULL,
+  actual_start_time timestamp NULL,
   actual_start_time_usec  int default 0,
   is_in_effect smallint default 0,
-  trigger_time timestamp NULL DEFAULT NULL,
+  trigger_time timestamp NULL,
   name TEXT character set latin1 default NULL,
   session_token int default NULL,
   PRIMARY KEY  (scheduleddowntime_id),
@@ -996,9 +996,9 @@
   max_check_attempts smallint default 0,
   state smallint default 0,
   state_type smallint default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   command_object_id bigint unsigned default 0,
   command_args TEXT character set latin1,
@@ -1199,7 +1199,7 @@
   servicestatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   service_object_id bigint unsigned default 0,
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   output TEXT character set latin1,
   long_output TEXT,
   perfdata TEXT character set latin1,
@@ -1209,19 +1209,19 @@
   should_be_scheduled smallint default 0,
   current_check_attempt smallint default 0,
   max_check_attempts smallint default 0,
-  last_check timestamp NULL DEFAULT NULL,
-  next_check timestamp NULL DEFAULT NULL,
+  last_check timestamp NULL,
+  next_check timestamp NULL,
   check_type smallint default 0,
-  last_state_change timestamp NULL DEFAULT NULL,
-  last_hard_state_change timestamp NULL DEFAULT NULL,
+  last_state_change timestamp NULL,
+  last_hard_state_change timestamp NULL,
   last_hard_state smallint default 0,
-  last_time_ok timestamp NULL DEFAULT NULL,
-  last_time_warning timestamp NULL DEFAULT NULL,
-  last_time_unknown timestamp NULL DEFAULT NULL,
-  last_time_critical timestamp NULL DEFAULT NULL,
+  last_time_ok timestamp NULL,
+  last_time_warning timestamp NULL,
+  last_time_unknown timestamp NULL,
+  last_time_critical timestamp NULL,
   state_type smallint default 0,
-  last_notification timestamp NULL DEFAULT NULL,
-  next_notification timestamp NULL DEFAULT NULL,
+  last_notification timestamp NULL,
+  next_notification timestamp NULL,
   no_more_notifications smallint default 0,
   notifications_enabled smallint default 0,
   problem_has_been_acknowledged smallint default 0,
@@ -1288,7 +1288,7 @@
 CREATE TABLE IF NOT EXISTS icinga_statehistory (
   statehistory_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  state_time timestamp NULL DEFAULT NULL,
+  state_time timestamp NULL,
   state_time_usec  int default 0,
   object_id bigint unsigned default 0,
   state_change smallint default 0,
@@ -1313,9 +1313,9 @@
 CREATE TABLE IF NOT EXISTS icinga_systemcommands (
   systemcommand_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
-  start_time timestamp NULL DEFAULT NULL,
+  start_time timestamp NULL,
   start_time_usec  int default 0,
-  end_time timestamp NULL DEFAULT NULL,
+  end_time timestamp NULL,
   end_time_usec  int default 0,
   command_line TEXT character set latin1,
   timeout smallint default 0,
@@ -1393,7 +1393,7 @@
   instance_id bigint unsigned default 0,
   endpoint_object_id bigint(20) unsigned DEFAULT '0',
   zone_object_id bigint(20) unsigned DEFAULT '0',
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   identity varchar(255) DEFAULT NULL,
   node varchar(255) DEFAULT NULL,
   is_connected smallint(6),
@@ -1425,7 +1425,7 @@
   zonestatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   zone_object_id bigint(20) unsigned DEFAULT '0',
-  status_update_time timestamp NULL DEFAULT NULL,
+  status_update_time timestamp NULL,
   parent_zone_object_id bigint(20) unsigned DEFAULT '0',
   PRIMARY KEY  (zonestatus_id)
 ) ENGINE=InnoDB COMMENT='Zone status';
--- a/lib/db_ido_mysql/schema/upgrade/2.2.0.sql
+++ b/lib/db_ido_mysql/schema/upgrade/2.2.0.sql
@@ -9,8 +9,8 @@
 
 ALTER TABLE `icinga_programstatus` ADD COLUMN `program_version` varchar(64) character set latin1 collate latin1_general_cs default NULL;
 
-ALTER TABLE icinga_contacts MODIFY alias TEXT character set latin1  default '';
-ALTER TABLE icinga_hosts MODIFY alias TEXT character set latin1  default '';
+ALTER TABLE icinga_contacts MODIFY alias TEXT character set latin1;
+ALTER TABLE icinga_hosts MODIFY alias TEXT character set latin1;
 
 ALTER TABLE icinga_customvariables ADD COLUMN is_json smallint default 0;
 ALTER TABLE icinga_customvariablestatus ADD COLUMN is_json smallint default 0;
--- a/lib/db_ido_mysql/schema/upgrade/2.4.0.sql
+++ b/lib/db_ido_mysql/schema/upgrade/2.4.0.sql
@@ -28,7 +28,7 @@
   zonestatus_id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   instance_id bigint unsigned default 0,
   zone_object_id bigint(20) unsigned DEFAULT '0',
-  status_update_time timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
+  status_update_time timestamp NOT NULL,
   parent_zone_object_id bigint(20) unsigned DEFAULT '0',
   PRIMARY KEY  (zonestatus_id)
 ) ENGINE=InnoDB COMMENT='Zone status';
--- a/lib/db_ido_mysql/schema/upgrade/2.6.0.sql
+++ b/lib/db_ido_mysql/schema/upgrade/2.6.0.sql
@@ -12,137 +12,137 @@
 -- -----------------------------------------
 
 ALTER TABLE icinga_acknowledgements
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_commenthistory
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN comment_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN expiration_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN deletion_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL,
+  MODIFY COLUMN comment_time timestamp NULL,
+  MODIFY COLUMN expiration_time timestamp NULL,
+  MODIFY COLUMN deletion_time timestamp NULL;
 
 ALTER TABLE icinga_comments
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN comment_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN expiration_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL,
+  MODIFY COLUMN comment_time timestamp NULL,
+  MODIFY COLUMN expiration_time timestamp NULL;
 
 ALTER TABLE icinga_conninfo
-  MODIFY COLUMN connect_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN disconnect_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_checkin_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN data_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN data_end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN connect_time timestamp NULL,
+  MODIFY COLUMN disconnect_time timestamp NULL,
+  MODIFY COLUMN last_checkin_time timestamp NULL,
+  MODIFY COLUMN data_start_time timestamp NULL,
+  MODIFY COLUMN data_end_time timestamp NULL;
 
 ALTER TABLE icinga_contactnotificationmethods
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_contactnotifications
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_contactstatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_host_notification timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_service_notification timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL,
+  MODIFY COLUMN last_host_notification timestamp NULL,
+  MODIFY COLUMN last_service_notification timestamp NULL;
 
 ALTER TABLE icinga_customvariablestatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL;
 
 ALTER TABLE icinga_dbversion
-  MODIFY COLUMN create_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN modify_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN create_time timestamp NULL,
+  MODIFY COLUMN modify_time timestamp NULL;
 
 ALTER TABLE icinga_downtimehistory
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN scheduled_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN scheduled_end_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN actual_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN actual_end_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN trigger_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL,
+  MODIFY COLUMN scheduled_start_time timestamp NULL,
+  MODIFY COLUMN scheduled_end_time timestamp NULL,
+  MODIFY COLUMN actual_start_time timestamp NULL,
+  MODIFY COLUMN actual_end_time timestamp NULL,
+  MODIFY COLUMN trigger_time timestamp NULL;
 
 ALTER TABLE icinga_eventhandlers
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_externalcommands
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL;
 
 ALTER TABLE icinga_flappinghistory
-  MODIFY COLUMN event_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN comment_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN event_time timestamp NULL,
+  MODIFY COLUMN comment_time timestamp NULL;
 
 ALTER TABLE icinga_hostchecks
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_hoststatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_check timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN next_check timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_state_change timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_hard_state_change timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_up timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_down timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_unreachable timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_notification timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN next_notification timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL,
+  MODIFY COLUMN last_check timestamp NULL,
+  MODIFY COLUMN next_check timestamp NULL,
+  MODIFY COLUMN last_state_change timestamp NULL,
+  MODIFY COLUMN last_hard_state_change timestamp NULL,
+  MODIFY COLUMN last_time_up timestamp NULL,
+  MODIFY COLUMN last_time_down timestamp NULL,
+  MODIFY COLUMN last_time_unreachable timestamp NULL,
+  MODIFY COLUMN last_notification timestamp NULL,
+  MODIFY COLUMN next_notification timestamp NULL;
 
 ALTER TABLE icinga_logentries
-  MODIFY COLUMN logentry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN logentry_time timestamp NULL,
+  MODIFY COLUMN entry_time timestamp NULL;
 
 ALTER TABLE icinga_notifications
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_processevents
-  MODIFY COLUMN event_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN event_time timestamp NULL;
 
 ALTER TABLE icinga_programstatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN program_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN program_end_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_command_check timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_log_rotation timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN disable_notif_expire_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL,
+  MODIFY COLUMN program_start_time timestamp NULL,
+  MODIFY COLUMN program_end_time timestamp NULL,
+  MODIFY COLUMN last_command_check timestamp NULL,
+  MODIFY COLUMN last_log_rotation timestamp NULL,
+  MODIFY COLUMN disable_notif_expire_time timestamp NULL;
 
 ALTER TABLE icinga_scheduleddowntime
-  MODIFY COLUMN entry_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN scheduled_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN scheduled_end_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN actual_start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN trigger_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN entry_time timestamp NULL,
+  MODIFY COLUMN scheduled_start_time timestamp NULL,
+  MODIFY COLUMN scheduled_end_time timestamp NULL,
+  MODIFY COLUMN actual_start_time timestamp NULL,
+  MODIFY COLUMN trigger_time timestamp NULL;
 
 ALTER TABLE icinga_servicechecks
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_servicestatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_check timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN next_check timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_state_change timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_hard_state_change timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_ok timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_warning timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_unknown timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_time_critical timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN last_notification timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN next_notification timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL,
+  MODIFY COLUMN last_check timestamp NULL,
+  MODIFY COLUMN next_check timestamp NULL,
+  MODIFY COLUMN last_state_change timestamp NULL,
+  MODIFY COLUMN last_hard_state_change timestamp NULL,
+  MODIFY COLUMN last_time_ok timestamp NULL,
+  MODIFY COLUMN last_time_warning timestamp NULL,
+  MODIFY COLUMN last_time_unknown timestamp NULL,
+  MODIFY COLUMN last_time_critical timestamp NULL,
+  MODIFY COLUMN last_notification timestamp NULL,
+  MODIFY COLUMN next_notification timestamp NULL;
 
 ALTER TABLE icinga_statehistory
-  MODIFY COLUMN state_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN state_time timestamp NULL;
 
 ALTER TABLE icinga_systemcommands
-  MODIFY COLUMN start_time timestamp NULL DEFAULT NULL,
-  MODIFY COLUMN end_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN start_time timestamp NULL,
+  MODIFY COLUMN end_time timestamp NULL;
 
 ALTER TABLE icinga_endpointstatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL;
 
 ALTER TABLE icinga_zonestatus
-  MODIFY COLUMN status_update_time timestamp NULL DEFAULT NULL;
+  MODIFY COLUMN status_update_time timestamp NULL;
 
 -- -----------------------------------------
 -- set dbversion
